File: sentimental-credit/credit.py
# steps:
# 1.Ask user for (valid) input
# 2.Luhn algorithm
# 3.Check the first 2 digits
# 4.Combine these, print results

from cs50 import get_int, get_string
import re

# ccnumber = ""
# while not re.match(r'^\d{13}$|^\d{15}$|^\d{16}$', ccnumber):
# ccnumber = get_string("Number:")
ccnumber = get_string("Number: ")
if len(str(ccnumber)) not in [13, 15, 16]:
    print("INVALID")
    exit()


def multiply_everyotherdigit_addthemtogether(ccnumber):
    i = 1
    sum_of_digits = 0
    cursor = 0
    cclength = len(str(ccnumber))
    while cclength - i - 1 >= 0:
        cursor = int(ccnumber[cclength - i - 1])
        if (2 * cursor) < 10:
            sum_of_digits += (2 * cursor)
        else:
            sum_of_digits += (2 * cursor - 9)
        i += 2
    print(sum_of_digits)
    return sum_of_digits


def sumof_notmultiplied(ccnumber):
    i = 0
    sum_of_digits2 = 0
    cursor = 0
    cclength = len(str(ccnumber))
    while cclength - i > 0:
        cursor = int(ccnumber[cclength - i - 1])
        sum_of_digits2 += cursor
        i += 2
    print(sum_of_digits2)
    return sum_of_digits2


total = (multiply_everyotherdigit_addthemtogether(ccnumber) + sumof_notmultiplied(ccnumber)) % 10
if total == 0:
    if (int(ccnumber[:2]) == 34 or int(ccnumber[:2]) == 37) and len(str(ccnumber)) == 15:
        print("AMEX")
    elif (51 <= int(ccnumber[:2]) <= 55) and len(str(ccnumber)) == 16:
        print("MASTERCARD")
    elif (int(ccnumber[0]) == 4) and (len(str(ccnumber)) == 13 or len(str(ccnumber)) == 16):
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>
long ccnumber;
long firstsecond;
long simplify(void);
int counter;
int y;
int main(void)
{
    do
    {
        ccnumber = get_long("Number: ");
    }
    while (ccnumber < 0);
    firstsecond = ccnumber;
    for (counter = 2; ccnumber > 100; counter++)
    {
        ccnumber = simplify();
    }

    if (ccnumber == 34 || ccnumber == 37)
    {
        if (counter == 15)
        {
            printf("AMEX\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }

    else if (50 < ccnumber && ccnumber < 56)
    {
        if (counter == 16)
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else if (39 < ccnumber && ccnumber < 50)
    {
        if (counter == 13 || counter == 16)
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}
long simplify(void)
{
    firstsecond = ccnumber / 10;
    return firstsecond;
}
